src/clients/detector.rs

```diff
@@ -108,6 +108,35 @@ impl DetectorClient {
             Err(error.into())
         }
     }
+
+    /// Invokes detectors implemented with the `/api/v1/text/context/doc` endpoint
+    pub async fn detect_for_context_documents(
+        &self,
+        model_id: &str,
+        request: ContextDocsDetectionRequest,
+    ) -> Result<Vec<DetectionResult>, Error> {
+        let client = self.client(model_id)?;
+        let url = client.base_url().as_str();
+        let response = client
+            .post(url)
+            .header(DETECTOR_ID_HEADER_NAME, model_id)
+            .json(&request)
+            .send()
+            .await?;
+        if response.status() == StatusCode::OK {
+            Ok(response.json().await?)
+        } else {
+            let code = response.status().as_u16();
+            let error = response
+                .json::<DetectorError>()
+                .await
+                .unwrap_or(DetectorError {
+                    code,
+                    message: "".into(),
+                });
+            Err(error.into())
```

Comment:
We should probably create a helper function to do this, to avoid the duplicated code in each method.

Reply:
Not essential for now as we should probably refactor the Detector impl anyway as discussed in chat. We can do this then.

Reply:
The only differences between `text_contents()`, `text_generation()` and `text_context_doc()` are:
1. The type of `request`
2. The "OK" variant type in return.
3. The detector endpoint

Let's open a separate issue and discuss some of these changes.